mh/pat_item.py

```diff
@@ -671,6 +673,257 @@ class LayerBinaryInfo(PatData):
         (0x03, "hunter_name")
     )
 
+class EquipmentSlot(object):
+    SIZE = 0xc
+
+    def __init__(self, type_id, slots, equipment_id, unk1, unk2, unk3):
+        self.type_id = type_id
+        self.slots = slots
+        self.equipment_id = equipment_id
+        self.unk1 = unk1
+        self.unk2 = unk2
+        self.unk3 = unk3
+
+    @staticmethod
+    def unpack(data, offset=0):
+        return EquipmentSlot(*struct.unpack_from(">BBHHIH", data, offset)[:6])
+
+    def pack(self):
+        return struct.pack(">BBHHIH", self.type_id, self.slots, self.equipment_id, self.unk1, self.unk2, self.unk3)
+
+class HunterSettings(object):
+    LENGTHS = (
+        0x02,  # 0) hr
+        0x01,  # 1) unk
+        0x01,  # 2) unk
+        0x01,  # 3) unk
+        0x01,  # 4) unk
+        0x02,  # 5) unk
+        0x04,  # 6) unk
+        0x04,  # 7) unk
+        0x0c,  # 8) equip 1
+        0x0c,  # 9) equip 2
+        0x0c,  # 10) equip 3
+        0x0c,  # 11) equip 4
+        0x0c,  # 12) equip 5
+        0x0c,  # 13) equip 6
+        0x0c,  # 14) equip 7
+        0x0c,  # 15) equip 8
+        0x0c,  # 16) equip 9
+        0x19,  # 17) unk
+        0x01,  # 18) unk
+        0x01,  # 19) unk
+        0x01,  # 20) unk
+        0x01,  # 21) unk
+        0x01,  # 22) unk
+        0x01,  # 23) unk
+        0x01,  # 24) unk
+        0x56,  # 25) profile
+        0x01,  # 26) title
+        0x01,  # 27) status
+        0x01,  # 28) unk
+        0x01,  # 29) hr limit type
+        0x01,  # 30) goal type
+        0x01,  # 31) seeking type
+        0x01,  # 32) server type
+        0x07  # 33) unk
+    )
+    SIZE = sum(LENGTHS)
+    
+    def __init__(self):
+        self.hr = 0
+        self.equipment_slot = [None for _ in range(9)]
+        self.profile = ''
+        self.title_id = 0
+        self.status_id = 0
+        self.hr_limit_type = 0
+        self.goal_type = 0
+        self.seeking_type = 0
+        self.server_type = 0
+    
+    def unpack(self, data, minBound, maxBound, offset=0):
```

Comment:
This logic is error prone. If possible, you should wait to have all the data before unpacking it. For instance, you can use a bytearray buffer:
```python
all_data = bytearray(LENGTHS)  # create a buffer of LENGTHS bytes
(...)
offset = 10  # Some offset
partial_data = ...
all_data[offset:offset+len(partial_data)] = partial_data  # Fill the buffer at offset
```